polygeist/include/polygeist/Passes/Passes.td

```diff
@@ -131,4 +131,10 @@ def ConvertPolygeistToLLVM : Pass<"convert-polygeist-to-llvm", "mlir::ModuleOp">
   ];
 }
 
+def ConvertToLLVMABI : Pass<"convert-to-llvm-abi", "mlir::ModuleOp"> {
+  let summary = "Convert MemRef ABI to LLVM Pointer ABI";
+  let constructor = "::mlir::polygeist::createConvertToLLVMABIPass()";
+  let dependentDialects = ["LLVM::LLVMDialect"];
+}
```

Comment:
I propose making this pass work on `gpu::GPUFuncOp` instead if we're just planning to modify those.

Reply:
We are planning to lower memrefs to match the usual colling conventions for any external function compiled by cgeist. We need to do this in order to support scenarios where the callee is compiled via cgeist and the caller might be compiled by any compiler (or viceversa). Technically functions that are private need not be converted, however it might be easier to just convert any function. 

Note that even external host functions need to match the usual calling conventions. So this is not a feature that is specific to the device module, it has to work for both the host and the device modules.